enterprise/frontend/src/embedding-sdk/cli/steps/pick-database-tables.ts

```diff
@@ -43,44 +46,90 @@ export const pickDatabaseTables: CliStepMethod = async state => {
     return [{ type: "error", message }, state];
   }
 
-  const tables: Table[] = [];
+  const tablesWithoutMetadata: Table[] = [];
 
-  // Fetch the database tables
-  for (const schemaKey of schemas) {
-    const url = `${instanceUrl}/api/database/${databaseId}/schema/${schemaKey}?include_hidden=true`;
+  try {
+    // Scan the database tables in each schemas
+    for (const schemaKey of schemas) {
+      const url = `${instanceUrl}/api/database/${databaseId}/schema/${schemaKey}?include_hidden=true`;
 
-    const res = await fetch(url, {
-      method: "GET",
-      headers: { "content-type": "application/json", cookie },
-    });
+      const res = await fetch(url, {
+        method: "GET",
+        headers: { "content-type": "application/json", cookie },
+      });
 
-    if (!res.ok) {
-      const message = `Cannot fetch database table from schema ${schemaKey}.`;
+      await propagateErrorResponse(res);
 
-      return [{ type: "error", message }, state];
+      const schemaTables: Table[] = await res.json();
+      tablesWithoutMetadata.push(...schemaTables);
     }
-
-    const schemaTables: Table[] = await res.json();
-
-    tables.push(...schemaTables);
+  } catch (error) {
+    return [cliError("Cannot scan database tables", error), state];
   }
 
-  if (tables.length === 0) {
+  if (tablesWithoutMetadata.length === 0) {
     spinner.fail();
 
     return [{ type: "error", message: "No tables found in database." }, state];
   }
 
   spinner.succeed();
 
-  const chosenTables = await checkbox({
-    validate: choices => choices.length > 0,
-    message: "Select the tables to embed:",
-    choices: tables.map(table => ({
+  const chosenTableIds = await checkbox({
+    validate: choices => {
+      if (choices.length === 0) {
+        return "Pick 1 - 3 tables to embed.";
+      }
+
+      if (choices.length > 3) {
+        return "You can only choose up to 3 tables.";
+      }
+
+      return true;
+    },
+    message: "Pick 1 - 3 tables to embed:",
+    choices: tablesWithoutMetadata.map(table => ({
       name: table.name,
-      value: table,
+      value: table.id,
     })),
   });
 
-  return [{ type: "done" }, { ...state, tables: chosenTables }];
+  const chosenTables: Table[] = [];
+
+  for (const tableId of chosenTableIds) {
+    const datasetQuery = {
+      type: "query",
+      database: databaseId,
+      query: { "source-table": tableId },
+    };
+
+    // The table's fields may still be syncing, so we retry a few times.
+    await retry(
+      async () => {
+        // Get the query metadata from a table
+        const res = await fetch(`${instanceUrl}/api/dataset/query_metadata`, {
```

Comment:
We need the query metadata to use in later steps, (create models and x-rays, setting up tenant isolation)